### PR TITLE
Max/code quality

### PR DESCRIPTION
**Pull request by max-hoffman**
Summary of changes:
- fixed linting errors (`poetry run mypy doltpy`)
- applied black syntax formatting (`poetry run black . --exclude tests -t 37`)
- added autogeneration of sphinx documentation (look at `scripts/docgen.sh`)
- added poetry dependency management starter, with release script stub (`pyproject.toml` and `scripts/release.sh`)
- added nox starter, which will add features on-top of GitHub actions matrices

TODOs:
- dependency management transition is not complete
- nox not fully implemented
- changes not added to CI

### CODE REVIEW

File: doltpy/cli/write/write.py [hunk 1]
```diff
@@ -88,27 +117,39 @@ def write_rows(dolt: Dolt,
     :param commit_date:
     :return:
     """
+
     def writer(filepath: str):
-        with open(filepath, 'w') as f:
-            fieldnames = set()
+        with open(filepath, "w") as f:
+            fieldnames: Set[str] = set()
```
Comment (oscarbatori) [inline doltpy/cli/write/write.py:123]: Is this type not inferred?

File: doltpy/etl/loaders.py [hunk 2]
```diff
@@ -294,11 +337,9 @@ def create_table_from_schema_import(repo: Dolt,
     _create_table_from_schema_import_helper(repo, table, pks, path, commit=commit, commit_message=commit_message)
 
 
-def create_table_from_schema_import_unique_key(repo: Dolt,
-                                               table: str,
-                                               path: str,
-                                               commit: bool = True,
-                                               commit_message: str = None):
+def create_table_from_schema_import_unique_key(
+    repo: Dolt, table: str, path: str, commit: bool = True, commit_message: str = None
```
Comment (oscarbatori) [inline doltpy/etl/loaders.py:341]: Seems this should be multi line?

**Review by oscarbatori — APPROVED** [review]

File: doltpy/etl/loaders.py (hunk 2 above)
Comment (max-hoffman) [reply doltpy/etl/loaders.py:341]: yep, i agree. black defaults to the simplest change to get a line syntax-compliant, can't always trust its output to be the nicest-looking

File: doltpy/cli/write/write.py (hunk 1 above)
Comment (max-hoffman) [reply doltpy/cli/write/write.py:123]: I only changed lines that had syntax errors, so I'm assuming no? There are some weird rules around what mypy considers OK or not. The python author decides the source code, PEP suggestions and mypy rules so we sort of just have to play ball

**Pull request merged**